src/common/systems/GameObjectController.cpp

```diff
@@ -145,4 +145,26 @@ std::pair<bool, uint32_t> GameObjectController::getGameOver() {
         }
     }
     return std::make_pair(false, std::numeric_limits<uint32_t>::max());
-}
\ No newline at end of file
+}
+
+std::pair<InputDef::ReliableInput, InputDef::UnreliableInput>
+GameObjectController::getBotMove(uint32_t bot_player_id) {
+    auto mines = m_gameObjectStore.getMines();
+    auto current_resources = m_resourceController.get(player_id);
+    auto bot_group_id = m_groupController.getGroupId(player_id);
+    Group& bot_group = m_groupController.getGroup(bot_group_id);
+    auto bot_group_center = bot_group.getCenter();
+    std::unordered_map<uint32_t, uint32_t> mine_id_to_resource_count;
+    for (auto mine : mines) {
+        auto mine_id = mine->getId();
+        mine_id_to_resource_count[mine_id] = m_resourceController.get(
+                mine_id, mine->getResourceType());
+    }
+
+    // required for greedy bot and most map aware bots:
+    // resources
+    // current location
+    // mine locations
+    return m_bot.getMove(BotStrategy::NearestGreedy, mines,
+            current_resources, bot_group_center, mine_id_to_resource_count);
```

Comment:
Instead of passing all this data as parameters, it would be cleaner if you made a `BotData` struct and put it there. 
I did something similar for `UIData`.

Do you think `BotData` should be a union of the data needed by all bots or should each `BotStrategy` have it's won struct?

Reply:
Im fine with it being a union unless we find it's annoying. What's the actual benefit of putting it in a struct?

Reply:
It's cleaner that way and someone can quickly see all the information needed for a bot.
I much prefer it.

Reply:
The tricky thing in this case is that the current inputs to this function are a union of what any bot strategy might desire. That means that if we struct it, we'd have to just pass bits and pieces down to strategies so they just have access to the parts they need. Personally I don't think putting it in a struct matters and it adds a layer of obfuscation when you try to understand what's in the struct